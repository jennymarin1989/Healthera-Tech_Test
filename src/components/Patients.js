import React, { Component} from 'react';
import Header from './Header';
import data from '../data.json';

class Patients extends Component {
	render(){
		return (
			<div>
				 <Header header = {"Patients"}/> 
				 <ul>
						 {
							data.map(function(patient){
								return <li> {patient.id}: {patient.name}  </li>  
							})
						 }
				 </ul>    
			</div>    
	)};  
};

export default Patients;