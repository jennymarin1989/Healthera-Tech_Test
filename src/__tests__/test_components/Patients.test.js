import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Patients from '../../components/Patients';

describe("Patients", () => {

    let patients = shallow(<Patients />)

   it('renders components correctly',() =>{
       expect(patients).toMatchSnapshot();
   }) ;
   
   it("renders patient name and id from data.jsn", () =>{
    let patientInfo = [
        {"id":1, "name":"Felix Smith"},
        {"id":2, "name":"Carlos Davis"},
        {"id":3, "name":"Maria Lopez"},
				{"id":4, "name":"Joseph martin"}
			]
        expect(toJson(patientInfo)).toMatchSnapshot();
    });

 describe("renders correctly", () => {
   it("render header component correctly", () => {
       expect(patients.find('Header').exists()).toBe(true);
   });
 });
 

});