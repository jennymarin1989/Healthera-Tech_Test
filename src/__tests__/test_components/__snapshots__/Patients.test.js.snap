// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patients renders components correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Patients />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          header="Patients"
/>,
        <ul>
          <li>
                     
                    1
                    : 
                    John Smith
                      
          </li>
          <li>
                     
                    2
                    : 
                    Claire Davis
                      
          </li>
          <li>
                     
                    3
                    : 
                    Mary Lopez
                      
          </li>
          <li>
                     
                    4
                    : 
                    Joseph Mull
                      
          </li>
</ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "header": "Patients",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
               
              1
              : 
              John Smith
                
</li>,
            <li>
               
              2
              : 
              Claire Davis
                
</li>,
            <li>
               
              3
              : 
              Mary Lopez
                
</li>,
            <li>
               
              4
              : 
              Joseph Mull
                
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                1,
                ": ",
                "John Smith",
                "  ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              1,
              ": ",
              "John Smith",
              "  ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                2,
                ": ",
                "Claire Davis",
                "  ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              2,
              ": ",
              "Claire Davis",
              "  ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                3,
                ": ",
                "Mary Lopez",
                "  ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              3,
              ": ",
              "Mary Lopez",
              "  ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                4,
                ": ",
                "Joseph Mull",
                "  ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              4,
              ": ",
              "Joseph Mull",
              "  ",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            header="Patients"
/>,
          <ul>
            <li>
                         
                        1
                        : 
                        John Smith
                          
            </li>
            <li>
                         
                        2
                        : 
                        Claire Davis
                          
            </li>
            <li>
                         
                        3
                        : 
                        Mary Lopez
                          
            </li>
            <li>
                         
                        4
                        : 
                        Joseph Mull
                          
            </li>
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "header": "Patients",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                 
                1
                : 
                John Smith
                  
</li>,
              <li>
                 
                2
                : 
                Claire Davis
                  
</li>,
              <li>
                 
                3
                : 
                Mary Lopez
                  
</li>,
              <li>
                 
                4
                : 
                Joseph Mull
                  
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  1,
                  ": ",
                  "John Smith",
                  "  ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                1,
                ": ",
                "John Smith",
                "  ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  2,
                  ": ",
                  "Claire Davis",
                  "  ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                2,
                ": ",
                "Claire Davis",
                "  ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  3,
                  ": ",
                  "Mary Lopez",
                  "  ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                3,
                ": ",
                "Mary Lopez",
                "  ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  4,
                  ": ",
                  "Joseph Mull",
                  "  ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                4,
                ": ",
                "Joseph Mull",
                "  ",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Patients renders patient name and id from data.jsn 1`] = `undefined`;
